atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -309,6 +311,48 @@ public void disableReadWriteConflictChecking(TableReference tableRef) {
         conflictDetectionManager.disableReadWriteConflict(tableRef);
     }
 
+    @Override
+    public ListenableFuture<Map<Cell, byte[]>> getAsync(TableReference tableRef, Set<Cell> cells) {
+        Timer.Context timer = getTimer("get").time();
+        checkGetPreconditions(tableRef);
+        if (Iterables.isEmpty(cells)) {
+            return Futures.immediateFuture(ImmutableMap.of());
+        }
+        hasReads = true;
+
+        ImmutableMap.Builder<Cell, byte[]> resultBuilder = ImmutableMap.builder();
```

Comment:
Same as above: eels like we should probably refactor some shared code here with #get instead of copying.